R/pkg_ref_cache_release_date.R

```diff
@@ -0,0 +1,31 @@
+#' Cache a List of Package Release Date from a Package Reference
+#'
+#' @inheritParams pkg_ref_cache
+#' @family package reference cache
+#'
+pkg_ref_cache.release_date <- function(x, name, ...) {
+  UseMethod("pkg_ref_cache.release_date")
+}
+
+
+pkg_ref_cache.release_date.pkg_remote <- function(x, name, ...) {
+
+  db  <- rvest::html_table(x$web_html)[[1]]
+  date <- db[grep("Publish",db[,1], ignore.case = TRUE) ,2]
+  date
+}
+
+
+pkg_ref_cache.release_date.pkg_install <- function(x, name, ...) {
+
+  colnames(x$description) <- tolower(colnames(x$description))
+  if (!"date" %in% colnames(x$description)){
+      return(NA)
+    }else{
+      return(x$description[, "date"])
+    }
+}
```

Comment:
This will modify the values of `x` (even outside this function scope), which can break other code that expects the colnames of `x$description` to keep the same upper/lower casing.
```suggestion
  if (!"Date" %in% colnames(x$description)) return(NA)
  x$description[, "Date"]
}
```